content/blog/2019/egress-traffic-control-in-istio-part-3/index.md

```diff
@@ -0,0 +1,136 @@
+
+Kubernetes provides a native solution for traffic control, and in particular, for control of egress traffic, through the [network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/).
+Using these network policies, cluster operators can configure which pods can access specific external services.
+Cluster operators can identify pods by pod labels, namespace labels, or by IP ranges. To specify the external services, cluster operators can use IP ranges, but cannot use domain names like `cnn.com`. This is because **Kubernetes network policies are not DNS-aware**.
+The network policies satisfy the first requirement since they can control any TCP traffic.
+Network policies only partially satisfy the third and the fourth requirements because cluster operators can specify policies
+per cluster or per pod but operators can't identify external services by domain names.
+Network policies only satisfy the fifth requirement if the attackers are not able to break from a malicious container into the Kubernetes
+node and interfere with the implementation of the policies inside said node.
+Lastly, network policies do satisfy the sixth requirement: Operators don't need to change the code or the
+container environment. In summary, we can say that Kubernetes Network Policies provide transparent, Kubernetes-aware egress traffic
+control, which is not DNS-aware.
+
+The second alternative predates the Kubernetes network policies. Using a **DNS-aware egress proxy or firewall** lets you
+configure applications to direct the traffic to the proxy and use some proxy protocol, for example,
+[SOCKS](https://en.wikipedia.org/wiki/SOCKS).
+Since operators must configure the applications, this solution is not transparent. Moreover, operators can't use
+pod labels or pod service accounts to configure the proxies because the egress proxies don't know about them. Therefore, **the egress proxies are not Kubernetes-aware** and can't fulfill the fourth requirement because
+egress proxies cannot enforce policies by source if a Kubernetes artifact specifies the source.
+In summary, egress proxies can fulfill the first, second, third and fifth requirements, but can't satisfy the fourth and
+the six requirements because they are not transparent and not Kubernetes-aware.
+
+## Advantages of Istio egress traffic control
+
+Istio egress traffic control is **DNS-aware**: you can define policies based on URLs or on wildcard domains like
+`*.ibm.com`. In this sense, it is better than Kubernetes network policies which are not DNS-aware.
+
+Istio egress traffic control is **transparent** with regard to TLS traffic, since Istio is transparent:
+you do not need to change the applications or to configure their containers.
+For HTTP traffic with TLS origination, you must configure the applications to use HTTP instead of HTTPS
+when the applications run with Istio sidecars injected.
```

Comment:
I mean "the applications that run as part of Istio service mesh". Maybe it is clear from the context, the stress here is that once you run an application as part of a service mesh (Istio sidecar is injected, manually or automatically), you have to configure that application explicitly to use HTTP instead of HTTPS.

Reply:
addressed in https://github.com/istio/istio.io/pull/4637/commits/8f7d21ef7e3300a9f4c9a4cc8c885dd6755dbbd9.